apis/login.js

```diff
@@ -0,0 +1,239 @@
+*                managerId:
+*                 type: string
+*                createdBy:
+*                 type: string
+*                updatedBy:
+*                 type: string 
+*     produces:
+*         application/json
+*     responses:
+*       200:
+*         description: user created successfully
+*         content:
+*           application/json:
+*            schema:
+*              type: object
+*              properties:
+*                _id:
+*                 type: string
+*                name:
+*                 type: string
+*                address:
+*                 type: object
+*                 properties:
+*                   flat_details:
+*                     type: string
+*                   area:
+*                      type: string
+*                   landmark:
+*                     type: string
+*                age:
+*                 type: integer
+*                mobile:
+*                 type: string
+*                gender:
+*                 type: string
+*                roles:
+*                  type: array
+*                  items:
+*                     type: integer
+*                email:
+*                 type: string
+*                password:
+*                 type: string
+*                avatar:
+*                 type: string
+*                managerId:
+*                 type: string
+*                createdBy:
+*                 type: string
+*                updatedBy:
+*                 type: string
+*                createdAt:
+*                 type: string
+*                 format: date
+*                updatedAt:
+*                 type: string
+*                 format: date
+*                isActive:
+*                 type: boolean
+*                isDeleted:
+*                 type: boolean
+*       400:
+*         description: Bad request
+*/
+
+const registerUser = async (req, res) => {
+  const validatePassword = (password) => {
+    const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+    return (regex.test(password));
+  }
+
+  if (!req.body.name.trim() || !req.body.address.trim() || !req.body.mobile.trim() || !req.body.gender.trim() || req.body.roles.length == 0 || !req.body.email.trim() || !req.body.password.trim() || req.body.age < 1) {
+    return res.status(400).send("name,address,age,mobile,gender,role,email and password are required fields");
+  }
+
+  if (req.body.age < 0 || req.body.age > 60) {
+    return res.status(400).send("age must be within 60");
+  }
+
+  if (!validator.isEmail(req.body.email)) {
+    return res.status(400).send("please enter valid email");
+  }
+ 
+  if (!validatePassword(req.body.password)) {
+    return res.status(400).send("Password must contain atleast one lower,one upper,one special character,one digit,no blank spaces and length must be between 8-20 characters");
+  }
+
+  const hashedPassword = await bcrypt.hash(req.body.password, 10);
+  const add = JSON.parse(req.body.address)
```

Comment:
The name of the variables should be meaningful. "Add" can have multiple meanings